XLS-0040d-decentralized-identity/README.md

```diff
@@ -0,0 +1,423 @@
+| `Account`        |:heavy_check_mark:  | `string`  |   `AccountID`   |
+
+Indicates the account which initiates the `DIDSet` transaction. This account MUST be a funded account on ledger.
+
+---
+| Field Name            |      Required?      | JSON Type | Internal Type   |
+| --------------------- | :-----------------: | :-------: | :-------------: |
+| `Fee` |  :heavy_check_mark: | `number`  |   `Amount`        |
+
+Indicates the fee that the account submitting this transaction is willing to pay.
+
+---
+| Field Name |      Required?      | JSON Type | Internal Type   |
+| -----------| :-----------------: | :-------: | :-------------: |
+| `URI`      |    | `string`  |   `BLOB`        |
+
+Indicates the `URI` field for this object. 
+ 
+---
+| Field Name |      Required?      | JSON Type | Internal Type   |
+| -----------| :-----------------: | :-------: | :-------------: |
+| `Data`     |                     | `string`  |   `BLOB`        |
+
+Indicates the `Data` field for this object.
+
+---
+
+| Field Name   |     Required?      | JSON Type | Internal Type 
+| -------------| :----------------: | :-------: | :-----------: 
+| `DIDDocument`       |                    | `string`  |   `BLOB`    |
+
+Indicates the DID document for this object.
+
+:notebook: If the `URI`, `Data`, or `DIDDocument` fields are missing, they will remain unchanged. However, if they contain an empty string, they will be deleted, otherwise, they will be updated.
+
+## 5.2. Read
+
+The initial step in utilizing a DID for an application may involve resolving the DID to access the underlying DID Document or attestations. This resolution process unveils the cryptographic material and service endpoints linked to the specific DID.
+
+In the case of XRPL DID, the `read` operation facilitates the resolution to the associated `DID` object. Upon successful resolution, applications can retrieve and utilize the relevant `URI` and/or `Data`, and/or `DIDDocument` fields contained within the `DID` object:
+ 
+Given the input DID for an account, follow these steps:
+
+- Retrieve the xrpl-specific-idstring of the DID.
+
+- If the xrpl-specific-idstring represents a public key:
+
+  - Compute the Account ID using the method described [here](https://xrpl.org/accounts.html#address-encoding).
+
+- Retrieve the contents of the corresponding `DID` object in its raw format using the XRPL's [`account_objects`]( https://xrpl.org/account_objects.html#) command. Use the `account` field to specify the Account ID to retrieve all objects owned by that account, including `DID` object in the raw ledger format. Or optionally use the `type` field to filter the results by `ledger_entry` type, i.e. `DID` to retrieve the contents of just the `DID` object.
+ 
+### Example
+Given the DID **did:xrpl:1:rDZdorj46soYhJKa26jm2cpJkgM3AfGhvu**, do the following:
+- Retrieve `rDZdorj46soYhJKa26jm2cpJkgM3AfGhvu`
+- Perform the `account_objects` method request to retrieve the contents of `DID` object:
```

Comment:
Thinking about this again, it would be better to do with `ledger_entry` and the param of `"did": "rDZdorj46soYhJKa26jm2cpJkgM3AfGhvu"` - there's no worry of dealing with pagination then.

Reply:
Updated the recommended method to `ledger_entry` and the `account_objects` method as an alternative.